### DISCUSSION TITLE
Could I get rid off the light gray line in the polygon face?

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [marcomusy/vedo#799] [Q&A] [answered]
Hi Marco,
As the following pic, there is one light gray diagonal line in each polygon face. Is it VTK's bug? And How can I get rid off them. Thanks!
BTW: I notice you have update VEDO website. It looks pretty good!
One little suggestion: If the function list of class is arranged alphabetically It will be better.
:)
![image](https://user-images.githubusercontent.com/116484589/215301561-cbdeda1b-fd39-4b57-9a31-8ae189bd50ca.png)

```
#encoding:utf-8
from vedo import *

#verts = [(0,0,0), (10,0,0), (10,0,20),(0,0,20) ,(0,0,0), (0,0,20),(10,10,20),(10,10,0) ,(0,0,0), (10,0,0),(10,-10*2**0.5/2,20),(0,-10*2**0.5/2,20)]
verts = [(0,0,0), (10,0,0), (10,0,20),(0,0,20) ,(-10,10,20),(-10,10,0),(-10,0,0),(-10,0,20)]#
faces=[(0,1,2,3),(0,5,4,3),]#,(0,3,7,6)]
mesh=Mesh([verts,faces]).lw(0.1)
mesh.compute_normals()
#verts_normals=[mesh.pointdata['Normals'][ivrt] for ivrt in mesh.points]
stps=mesh.points()
endps=mesh.pointdata['Normals']
itemp=2
arrows=Arrows(stps,stps+mesh.pointdata['Normals'])
labs = mesh.labels('id').c('black')
itemp=10
#~~~
mesh2=mesh.clone().z(20).lw(0.1)
labs2 = mesh2.labels('id').c('black')
#~~~
cells=mesh.faces()
pt_normals=mesh.pointdata['Normals']
tcell=cells[0]
count=0
extrude_faces=[]
for icell in cells:
	share_pt=[]
	for ipt in icell:
		if ipt in tcell:share_pt.append(ipt)
	if len(share_pt)==2:
		pass
#~~~
show(mesh,mesh2,interactive=True)
```

**Comment by marcomusy** [comment, marked as answer]
> As the following pic, there is one light gray diagonal line in each polygon face. Is it VTK's bug? And How can I get rid off them.

unfortunately this is a recent vtk bug (for sure it was not there in version 8.x) maybe you can try 9.0.3 (i dont remember if the bug was already there)

> BTW: I notice you have update VEDO website. It looks pretty good!
> One little suggestion: If the function list of class is arranged alphabetically It will be better.

done :)

**Reply by ZiguoAtGitHub** [reply]
>> > As the following pic, there is one light gray diagonal line in each polygon face. Is it VTK's bug? And How can I get rid off them.
 
`unfortunately this is a recent vtk bug (for sure it was not there in version 8.x) maybe you can try 9.0.3 (i dont remember if the bug was already there).`
**umh..... But here is one funny trick after I tried several times: the light gray line would disappear when setting the lw over 1.4.😂**

> > BTW: I notice you have update VEDO website. It looks pretty good!
> > One little suggestion: If the function list of class is arranged alphabetically It will be better.
> 

 ` done :)`
**Good Job!👍**
